syntax = "proto3";

service ChatService  {

    rpc getUsers(Request) returns (stream Credentials) {}
    rpc getInbox(Request) returns (stream Message) {}
    rpc CreateAccount(stream Credentials) returns (AccountStatus) {}
    rpc SendChat(stream Message) returns (stream MessageStatus) {}
    rpc DeleteAccount(Credentials) returns (AccountStatus) {}
}



message Message {

    string content = 1;
    string sent_time = 2;
    string recipient = 3;
}

message Credentials {
    string username = 1;
    string password = 2; 

}
message Request {
    int32 request_status = 1;
}

message MessageStatus {
    int32 message_status = 1; 
}
message AccountStatus {
    bool AccountStatus = 1; 
}